proto/vulcanize/registry/v1beta1/attributes.proto

```diff
@@ -5,29 +5,206 @@ import "gogoproto/gogo.proto";
-  string build_artifact_cid = 3 [(gogoproto.moretags) = "json:\"buildArtifactCID\" yaml:\"buildArtifactCID\""];
-  string tls_cert_cid       = 4 [(gogoproto.moretags) = "json:\"TLSCertCID\" yaml:\"TLSCertCID\""];
+  HashReference repo_hash_reference = 2
+      [(gogoproto.moretags) = "json:\"repoHashReference\" yaml:\"repoHashReference\""];
+  HashReference build_artifact_hash_ref = 3 [(gogoproto.moretags) = "json:\"buildArtifactHashRef\" yaml:\"buildArtifactHashRef\""];
+  HashReference tls_cert_hash_ref       = 4 [(gogoproto.moretags) = "json:\"tlsCertHashRef\" yaml:\"tlsCertHashRef\""];
   string type               = 5 [(gogoproto.moretags) = "json:\"type\" yaml:\"type\""];
   string version            = 6 [(gogoproto.moretags) = "json:\"version\" yaml:\"version\""];
+}
+
+message GitRepository{
+  string name =1  [(gogoproto.moretags) = "json:\"name\" yaml:\"name\""];
+  string repo_reference=2 [(gogoproto.moretags) = "json:\"repo_reference\" yaml:\"repo_reference\""];
+  string description=3  [(gogoproto.moretags) = "json:\"description\" yaml:\"description\""];
+  string version            = 4 [(gogoproto.moretags) = "json:\"version\" yaml:\"version\""];
+}
+
+message Binary{
+  HashReference hash_reference=1;
+  string targeted_arch=2;
+  string runtime_version=3;
+  HashReference repo_hash_reference=4;
+  string version=5;
+  string type=6;
+}
+
+message DockerImage{
+  string image_id=1;
+  HashReference binary_hash_reference=2;
+  HashReference repo_hash_reference=3;
+  string version=4;
+  string type=5;
+}
+
+message WatcherRegistrationRecord{
+  WatcherMetadata metadata =1;
+  HashReference repo_hash_reference=2;
+  WASMBinary wasm=3;
+  string version=4;
+  string type=5;
+
+  message WatcherMetadata{
+    string version=1;
+    HashReference chain_hash_reference=2;
+  }
+
+  message WASMBinary{
+    HashReference hash_reference=1;
+    WASMBinaryMetadata metadata=2;
+  }
+  
+  message WASMBinaryMetadata{
+    string compiler_version=1;
+    string execution_engine_version=2;
+  }
+}
+
+message ResponderContract{
+  HashReference service_provider_hash_ref=1;
+  HashReference auction_hash_ref=2;
+  HashReference watcher_hash_ref=3;
+  string version=4;
+  string type=5;
+}
+
+message JSPackage{
+  HashReference repo_hash_reference=1;
+  HashReference js_package_hash_ref=2;
+  string version=3;
+  string type=4;
+  string name =5;
+}
+
+message ChainRegistrationRecord{
+  string name=1;
+  repeated string ipld_types=2;
+  string type=3;
+  string version=4;
+  string chain_id=5;
+  string netwrok_id=6;
+  string genesis_hash=7;
+}
+
+message AppRecord{
+  string id=1;
+  string name=2;
+  string package=3;
+  string type=4;
+  string version=5;
+  string build=6;
+  string display_name=7;
+}
+
+message BotRecord{
+  string type=1;
+  string version=2;
+  string display_name=3;
+  string name =4;
+  Packages packages=5;
+
+  message Packages{
+
+    // method 1
```

Comment:
@i-norden Can you suggest a preffered method here?

Reply:
Same here unfortunately, not familiar with these types and isn't obvious to me what `ref` and `os` are supposed to represent in these mappings. Going to ask Ashwin.